.gitlab-ci.yml

```diff
@@ -12,3 +11,32 @@ build and push emulator image:
   script:
     - docker-compose -f ./docker-compose.yml build
     - docker-compose -f ./docker-compose.yml push
+
+build test:
+  stage: test
+  image: golang
+  variables:
+    TEST: "build-local"
+  script:
+    - go version
+    - go build .
+    - go test -v ./...
+
+go lint:
+  stage: test
+  image: golang
+  variables:
+    TEST: "golangci"
+  script:
+    - curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.32.0
+    - golangci-lint run
+
+version check:
+  stage: test
+  image: golang
+  variables:   
+    TEST: "version-check"
+  script:
+    - diff -u VERSION <(grep "var version = " trezord.go | cut -f 2 -d '"')
+
+#Gitlab does not support xgo building without hacking the base xgo image, for now builds are done maully until we find a better solution.
```

Comment:
Is there any issue that tracks the ability to cross-compile go in GitLab CI? Would be cool to link it here.

Reply:
The issue is not cross compile of go on gitlabci but rather docker tool xgo which relays on running docker inside its docker image that is being pulled to gitlab. There are workarounds, the best one i have found is [this](https://github.com/vdovhanych/xgo) which then in turn helps you build [this image](https://github.com/vdovhanych/xgo-with-docker) but the project is not being worked on and we would need to update it and maintain it ourselves.

Reply:
I'm wondering if running this particular job on a runner dirrectly (and not inside docker) would help here. But I admit I have no idea if that's even a viable approach here.

I'll resolve this to not hider mergning this PR. This isn't anything crucial.